src/ServerRequestFactory.php

```diff
@@ -198,21 +198,18 @@ public static function marshalHeaders(array $server)
     {
         $headers = [];
         foreach ($server as $key => $value) {
-            if ($value && strpos($key, 'HTTP_') === 0) {
-                $name = strtr(substr($key, 5), '_', ' ');
-                $name = strtr(ucwords(strtolower($name)), ' ', '-');
+            if ($value) {
```

Comment:
I read it in following way:
* we are only interested if there is some value
* then we are curious what key we have
 * if key starts with `HTTP_` - drop prefix
 * if key starts with `CONTENT_` - take it as is
 * ignore key otherwise
* normalize key (common for both cases for simplicity)
 * replace `_` with `-` in key
 * make key lowercase

Added cyclomatic complexity is only because we've decoupled common operations for `HTTP_*` and `CONTENT_*` keys, otherwise it is the same. I thought there is no reason to duplicate 3 exactly the same lines in 2 neighbor blocks.